okio/src/main/java/okio/RealBufferedSource.java

```diff
@@ -368,6 +368,22 @@
     }
   }
 
+  @Override public boolean rangeEquals(long offset, ByteString bytes) throws IOException {
+    return rangeEquals(offset, bytes, 0, bytes.size());
+  }
+
+  @Override public boolean rangeEquals(long offset, ByteString bytes, int bytesOffset, int count)
+      throws IOException {
+    if (closed) throw new IllegalStateException("closed");
+
+    for (int i = 0; i < count; i++) {
+      long bufferOffset = offset + i;
+      if (!request(bufferOffset + 1)) return false;
```

Comment:
Yeah, After thinking about it for 3 seconds I think this is actually necessary for correctness. Otherwise this method could block when it isn’t supposed to.